File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) return 0;

        int i = 0;
        int r = 1;

        while (r < nums.length) {
            if (nums[r] != nums[i]) {
                i++;
                nums[i] = nums[r];
            }
            r++;
        }

        return i + 1;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.py
def findStart(nums):
    lo = 0
    hi = len(nums) - 1

    while lo < hi:
        mid = (lo + hi) // 2

        if nums[mid] > nums[hi]:
            lo = mid + 1
        else:
            hi = mid

    return lo

def rotate(i:int, s:int, n:int):
    return (i + s) % n


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        s = findStart(nums)
        n = len(nums)
        
        lo = 0
        hi = len(nums) - 1

        while lo <= hi:
            mid = (lo + hi) // 2
            i = rotate(mid, s, n)

            if nums[i] == target:
                return i
            elif nums[i] < target:
                lo = mid + 1
            else:
                hi = mid - 1
    
        return -1
        